diff --git a/src/filter.c b/src/filter.c
index c075e5c6..2fbf74f9 100644
--- a/src/filter.c
+++ b/src/filter.c
@@ -1081,7 +1081,7 @@ trackpoint_accel_profile(struct motion_filter *filter,
 	/* This is almost the equivalent of the xserver acceleration
 	   at sensitivity 128 and speed 0.0 */
 	factor = delta * accel_filter->incline + accel_filter->offset;
-	factor = min(factor, max_accel);
+	/* factor = min(factor, max_accel); */
 
 	return factor;
 }
