diff --git a/src/filter.c b/src/filter.c
index a1a00f04..c075e5c6 100644
--- a/src/filter.c
+++ b/src/filter.c
@@ -1147,7 +1147,7 @@ trackpoint_accelerator_filter(struct motion_filter *filter,
 	coords.x = scaled.x * f;
 	coords.y = scaled.y * f;
 
-	coords = trackpoint_clip_to_max_delta(accel_filter, coords);
+	/* coords = trackpoint_clip_to_max_delta(accel_filter, coords); */
 
 	return coords;
 }
